### PR TITLE
Move VisaDetail to packages/components

### PR DESCRIPTION
**Pull request by KarinaDavtyan**

### CODE REVIEW

File: apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js [hunk 1]
```diff
@@ -1,67 +1,31 @@
...
+|};
 
 const isVisaRequired = (
   passengers: $ReadOnlyArray<?VisaRequiredType>,
 ): boolean => passengers.some(passenger => passenger?.visaRequired);
 
-const VisaDetail = (props: Props) => {
+const Visa = (props: Props) => {
   const passengers = props.data?.passengers;
   const visaRequired = passengers && isVisaRequired(passengers);
```
Comment (tbergq) [inline apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js]: I know this is from a previous PR, but I think we should rather handle the case where passengers is null/undefined in `isVisaRequired`

```
const isVisaRequired = (
  passengers: ?$ReadOnlyArray<?VisaRequiredType>,
): boolean => {
if (passengers == null) return false;
return passengers.some(passenger => passenger?.visaRequired);	): boolean => passengers.some(passenger => passenger?.visaRequired);
}
```

then just call it like `isVisaRequired(props.data?.passengers)`

File: packages/components/src/visaDetail/VisaDetail.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { StyleSheet, Text, Icon } from '@kiwicom/universal-components';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+
+type Props = {
+  +visaRequired: ?boolean,
+};
```
Comment (tbergq) [inline packages/components/src/visaDetail/VisaDetail.js]: Type should be exact

File: apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js [hunk 3]
```diff
@@ -1,67 +1,31 @@
...
   +visaRequired: ?boolean,
-};
+|};
 
 const isVisaRequired = (
   passengers: $ReadOnlyArray<?VisaRequiredType>,
 ): boolean => passengers.some(passenger => passenger?.visaRequired);
 
-const VisaDetail = (props: Props) => {
+const Visa = (props: Props) => {
```
Comment (jvaclavik) [inline apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js]: Also file should be renamed

Comment (jvaclavik) [reply apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js]: There are also some names like 
```
import { VisaDetail } from '@kiwicom/margarita-components';
import type { VisaDetail_data as VisaDetailType } from './__generated__/VisaDetail_data.graphql';

...
  +data: ?VisaDetailType,

```

With VisaDetail, it should be consistent.

File: apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js (hunk 1 above)
Comment (KarinaDavtyan) [reply apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js]: Changes, looks better now

File: packages/components/src/visaDetail/VisaDetail.js (hunk 2 above)
Comment (KarinaDavtyan) [reply packages/components/src/visaDetail/VisaDetail.js]: Fixed!

File: apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js (hunk 3 above)
Comment (KarinaDavtyan) [reply apps/core/scenes/bookingDetail/menuGroups/Passengers/VisaDetail.js]: Now its `VisaInfoSummary` as its info for all the passengers in the booking, and the component itself called `VisaInfo`

**Pull request merged**